src/kirby/components/loading-overlay/loading-overlay.component.tns.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, OnInit, Input, ViewChild, ElementRef, AfterViewInit } from '@angular/core';
+import { isAndroid } from 'tns-core-modules/ui/page/page';
+import { LayoutBase, View } from 'tns-core-modules/ui/layouts/layout-base';
+import { BehaviorSubject } from 'rxjs';
+
+@Component({
+  selector: 'kirby-loading-overlay',
+  templateUrl: './loading-overlay.component.html',
+  styleUrls: ['./loading-overlay.component.scss'],
+})
+export class LoadingOverlayComponent implements OnInit, AfterViewInit {
+  private isLoadingSubject = new BehaviorSubject(false);
+
+  private _isLoading: boolean;
+  public get isLoading(): boolean {
+    return this._isLoading;
+  }
+  @Input()
+  public set isLoading(v: boolean) {
+    this._isLoading = v;
+    this.isLoadingSubject.next(v);
+  }
+
+  @Input() public showBackdrop = true;
+
+  @ViewChild('overlayWrapperRef')
+  wrapperGrid: ElementRef;
+
+  constructor() {}
+
+  ngOnInit() {}
+
+  ngAfterViewInit(): void {
+    this.isLoadingSubject.subscribe((isLoading) => {
+      this.setControlInteractionState(<View>this.wrapperGrid.nativeElement, !isLoading);
+    });
+  }
+
+  // recursively enable/disable interactions for controls as Android is not blocking tap events on overlay
+  // https://stackoverflow.com/questions/40988124/nativescript-disable-all-controls-while-activityindicator-is-shown/42331788#42331788
+  setControlInteractionState(view: View, isEnabled: boolean): void {
+    debugger;
```

Comment:
🔫  peww peww, die `debugger` statement.

Reply:
Done, can we automate this in the future?

Reply:
Yes we could write a linting rule to prevent this.

Reply:
We don't need to write them even. They are already created:
https://palantir.github.io/tslint/rules/no-debugger/
I might create a PR for how we fix all of this. We can discuss this in the Kirby guild meetings.